src/code/z_skelanime.c

```diff
@@ -1,176 +1,1675 @@
+        if (limbDlist[0] != NULL) {
+            Matrix_ToMtx(*limbMatricies, "../z_skelanime.c", 1242);
+            gSPMatrix(gfxCtx->polyOpa.p++, *limbMatricies, G_MTX_LOAD);
+            gSPDisplayList(gfxCtx->polyOpa.p++, &limbDlist[0]);
+            (*limbMatricies)++;
+        } else {
+            if (&limbDlist[1] != 0) {
+                Matrix_ToMtx(*limbMatricies, "../z_skelanime.c", 1249);
+                (*limbMatricies)++;
+            }
+        }
+    }
+
+    if (appendDlistFunc != NULL) {
+        appendDlistFunc(globalCtx, limbIndex, &limbDlist[1], &limbRot, actor);
+    }
+
+    if (limbEntry->firstChildIndex != LIMB_DONE) {
+        SkelAnime_DrawLimbSV(globalCtx, limbEntry->firstChildIndex, limbTable, actorDrawTable, updateMtxFunc,
+                             appendDlistFunc, actor, limbMatricies);
+    }
+
+    Matrix_Pull();
+
+    if (limbEntry->nextLimbIndex != LIMB_DONE) {
+        SkelAnime_DrawLimbSV(globalCtx, limbEntry->nextLimbIndex, limbTable, actorDrawTable, updateMtxFunc,
+                             appendDlistFunc, actor, limbMatricies);
+    }
+
+    func_800C6B54(sp34, globalCtx->state.gfxCtx, "../z_skelanime.c", 1265);
+}
+#else
+#pragma GLOBAL_ASM("asm/non_matchings/code/z_skelanime/SkelAnime_DrawLimbSV.s")
+#endif
+
+#ifdef NON_MATCHING
+void SkelAnime_DrawSV(GlobalContext* globalCtx, u32* limbTable, Vec3s* actorDrawTable, u32 dListCount,
+                      SkelAnime_LimbUpdateMatrix updateMtxFunc, SkelAnime_LimbAppendDlist appendDlistFunc,
+                      Actor* actor) {
+    SkelLimbEntry* rootLimb;
+    Gfx* skelDlist[2];
+    Vec3f pos;
+    Vec3s rot;
+    GraphicsContext* gfxCtx;
+    Mtx* limbMatricies;
+    Gfx* gfx[5];
+
+    limbMatricies = Graph_Alloc(globalCtx->state.gfxCtx, dListCount << 6);
+
+    if (limbTable == NULL) {
+        osSyncPrintf(VT_FGCOL(RED));
+        osSyncPrintf("Si2_draw_SV():skelがNULLです。\n"); // Si2_draw_SV (): skel is NULL.
+        osSyncPrintf(VT_RST);
+        return;
+    }
+
+    gfxCtx = globalCtx->state.gfxCtx;
+
+    func_800C6AC4(gfx, globalCtx->state.gfxCtx, "../z_skelanime.c", 1294);
+
+    gSPSegment(gfxCtx->polyOpa.p++, 0xD, limbMatricies);
+    Matrix_Push();
+    rootLimb = SEGMENTED_TO_VIRTUAL(*limbTable);
+
+    pos.x = (f32)actorDrawTable[0].x;
+    pos.y = (f32)actorDrawTable[0].y;
+    pos.z = (f32)actorDrawTable[0].z;
+
+    rot = actorDrawTable[1];
+
+    skelDlist[0] = skelDlist[1] = rootLimb->displayLists[0];
+
+    if ((updateMtxFunc == NULL) || (updateMtxFunc(globalCtx, 1, &skelDlist[0], &pos, &rot, actor) == 0)) {
+        Matrix_TranslateThenRotateZYX(&pos, &rot);
+        if (skelDlist[0] != NULL) {
+            Matrix_ToMtx(limbMatricies, "../z_skelanime.c", 1327);
+            gSPMatrix(gfxCtx->polyOpa.p++, limbMatricies, G_MTX_LOAD);
+            gSPDisplayList(gfxCtx->polyOpa.p++, skelDlist[0]);
+            limbMatricies++;
+        } else {
+            if (skelDlist[1] != 0) {
```

Comment:
```suggestion
            if (skelDlist[1] != NULL) {
```
I would also make this into a simple else if, and anywhere else this pattern is used.